sql/wsrep_mysqld.cc

```diff
@@ -2345,6 +2364,20 @@ int wsrep_to_isolation_begin(THD *thd, const char *db_, const char *table_,
   return ret;
 }
 
+bool wsrep_thd_is_in_to_isolation(THD *thd, bool flock)
+{
+  bool status = false;
+  if (thd)
+  {
+    if (flock) mysql_mutex_lock(&thd->LOCK_thd_data);
+
+    status = thd->wsrep_skip_wsrep_hton;
+
+    if (flock) mysql_mutex_unlock(&thd->LOCK_thd_data);
```

Comment:
Just a thought, (not a blocker).

If we are protecting `thd->wsrep_skip_wsrep_hton` with `thd->LOCK_thd_data`, then we probably should acquire it here as well. 

https://github.com/percona/percona-xtradb-cluster/blob/5.7/sql/wsrep_hton.cc#L92-L93

Or may be it is not required since writing and reading happens in the same thread.